packages/ui-extensions/src/surfaces/checkout/api/shared.ts

```diff
@@ -148,3 +148,34 @@ export interface ShippingAddress extends MailingAddress {
    */
   oneTimeUse?: boolean;
 }
+
+type MaybePromise<T> = T extends Promise<any> ? T : T | Promise<T>;
+type Subscriber<T> = (value: T) => unknown;
+type RemoteSubscribeResult<T> = [() => void, T];
+
+/**
+ * A remote-subscribable object managed on the main thread.
+ *
+ * Example: Checkout uses this to manage the state of an object and
+ * communicate state changes to extensions running in a sandboxed web worker.
+ */
+export interface RemoteSubscribable<T> {
+  readonly initial: T;
+  subscribe(subscriber: Subscriber<T>): MaybePromise<RemoteSubscribeResult<T>>;
+}
+
+/**
+ * A remote-subscribable object exposed to a sandboxed web worker.
+ *
+ * Example: extensions use this to get updates about an object
+ * managed by Checkout on the main thread.
+ *
+ * This interface is compatible with Preact's ReadonlySignal:
+ * https://github.com/preactjs/signals/blob/a023a132a81bd4ba4a0bebb8cbbeffbd8c8bbafc/packages/core/src/index.ts#L700-L709
```

Comment:
This looks like it's referencing a specific commit, should it be main or something like that? 
```suggestion
 * https://github.com/preactjs/signals/blob/main/packages/core/src/index.ts#L700-L709
```

Reply:
thinking it could change line positions on a specific branch like main, right?

Reply:
Yeah, referencing the commit is safer in case the code gets moved around. It's a pretty stable interface so it probably won't change.